.github/workflows/binaries.yml

```
@@ -0,0 +1,34 @@
+name: Binaries
+
+on:
+  workflow_dispatch:
+
+jobs:
+  binary:
+    name: Create
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: macos-13
+            name: mac-intel
+          - os: macos-14
+            name: mac-arm
+          - os: ubuntu-latest
+            name: linux-intel
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: cachix/install-nix-action@v22
+      - run: nix build
+      - uses: actions/upload-artifact@v4
+        with:
+          path: result/bin/dune
+          name: dune-${{ matrix.name }}
+  merge:
```

Since I was surprised what it merges (turns out: merge not the git sense but rather a "merge actions into one" way)
```suggestion
  upload:
```

---

the first build uploads a single-arch zip, and the this one uploads a combination zip, so i used `combine:`